### PR TITLE
Fix: ps_imageslider - Images with invalid name are saved but not displaying correctly

### LINKED ISSUE #37452: ps_imageslider - Images with invalid name are saved but not displaying correctly
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I uploaded an image with a specific name "-30% blackweek.jpg" but it is not displayed correctly.

Image

Image

### Expected behavior

Image is displayed correctly.
Invalid characters are stripped or replaced with valid character.

### Steps to reproduce

1. Go to module "Carousel" (ps_imageslider)
2. Create a new slide and upload an image named "-30%.jpg".
3. Save

### PrestaShop version(s) where the bug happened

8.2.0

### PHP version(s) where the bug happened

8.1.30

### If your bug is related to a module, specify its name and its version

ps_sliderimage v3.2.1

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by Codencode**
Converted the file name using `Tools::str2url()`

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | I uploaded an image with a specific name "-30% blackweek.jpg" but it is not displayed correctly.
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/37452
| Sponsor company   | @codencode
| How to test?      | Uploading a file that contains the character: %

### CODE REVIEW

**Review by boherm — APPROVED** [review]

**Comment by boherm** [comment]
Thanks @Codencode!

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
